File: workshop_content/KI_WORKSHOP/VB.Net_Vergleichsgruppe/Provisionierung/Domain/ProvisionsEingabe.vb
Namespace Domain

    ''' <summary>
    ''' Immutable Input-Objekt für die Provisionsberechnung.
    ''' Enthält alle Eingabedaten ohne UI-Abhängigkeiten.
    ''' </summary>
    Public Structure ProvisionsEingabe

        Public ReadOnly Property Umsatz As Decimal
        Public ReadOnly Property Kategorie As Kategorie
        Public ReadOnly Property IstVipKunde As Boolean
        Public ReadOnly Property IstJubilaeumsAktion As Boolean

        Public Sub New(umsatz As Decimal, kategorie As Kategorie, istVipKunde As Boolean, istJubilaeumsAktion As Boolean)
            Me.Umsatz = umsatz
            Me.Kategorie = kategorie
            Me.IstVipKunde = istVipKunde
            Me.IstJubilaeumsAktion = istJubilaeumsAktion
        End Sub

    End Structure

End Namespace

File: workshop_content/KI_WORKSHOP/C#_Vergleichsgruppe/Provisionierung/Domain/ProvisionsEingabe.cs
namespace Provisionierung.Domain;

/// <summary>
/// Immutable Input-Objekt für die Provisionsberechnung.
/// Enthält alle Eingabedaten ohne UI-Abhängigkeiten.
/// </summary>
public readonly struct ProvisionsEingabe
{
    public decimal Umsatz { get; }
    public Kategorie Kategorie { get; }
    public bool IstVipKunde { get; }
    public bool IstJubilaeumsAktion { get; }

    public ProvisionsEingabe(decimal umsatz, Kategorie kategorie, bool istVipKunde, bool istJubilaeumsAktion)
    {
        Umsatz = umsatz;
        Kategorie = kategorie;
        IstVipKunde = istVipKunde;
        IstJubilaeumsAktion = istJubilaeumsAktion;
    }
}
